src/evaluators/BinaryExpression.js

```diff
@@ -230,6 +230,8 @@ export function computeBinary(
       }
       // return or throw completion
       if (completion instanceof AbruptCompletion) throw completion;
+      if (completion instanceof NormalCompletion && !(completion instanceof PossiblyNormalCompletion))
```

Comment:
At this point (at least in master) completion is guaranteed be an instance of Value.